application/forms/Jobs/JobConfigForm.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+// Icinga Web 2 X.509 Module | (c) 2018 Icinga GmbH | GPLv2
+
+namespace Icinga\Module\X509\Forms\Jobs;
+
+use Exception;
+use Icinga\Authentication\Auth;
+use Icinga\Module\X509\Common\Database;
+use Icinga\Module\X509\Model\X509Job;
+use Icinga\Web\Notification;
+use ipl\Html\HtmlElement;
+use ipl\Stdlib\Str;
+use ipl\Validator\CallbackValidator;
+use ipl\Validator\CidrValidator;
+use ipl\Web\Compat\CompatForm;
+use ipl\Web\Url;
+
+/**
+ * Create, update and delete jobs
+ */
+class JobConfigForm extends CompatForm
+{
+    use Database;
+
+    /** @var X509Job */
+    protected $job;
+
+    public function __construct(X509Job $job = null)
+    {
+        $this->job = $job;
+    }
+
+    protected function isUpdating(): bool
+    {
+        return Url::fromRequest()->getPath() === 'x509/job/update';
+    }
+
+    public function hasBeenSubmitted(): bool
+    {
+        $button = $this->getPressedSubmitElement()->getName();
```

Comment:
This can't work correctly as `getPressedSubmitElement()` may return `null`. Also the base class' method calls `hasBeenSent()` first for early exit which makes sense to keep.